tensorflow/python/tools/saved_model_cli.py

```diff
@@ -137,24 +146,121 @@ def _show_inputs_outputs(saved_model_dir, tag_set, signature_def_key, indent=0):
       meta_graph_def, signature_def_key)
 
   indent_str = '  ' * indent
+
   def in_print(s):
     print(indent_str + s)
 
   in_print('The given SavedModel SignatureDef contains the following input(s):')
   for input_key, input_tensor in sorted(inputs_tensor_info.items()):
     in_print('  inputs[\'%s\'] tensor_info:' % input_key)
-    _print_tensor_info(input_tensor, indent+1)
+    _print_tensor_info(input_tensor, indent + 1)
 
   in_print('The given SavedModel SignatureDef contains the following '
            'output(s):')
   for output_key, output_tensor in sorted(outputs_tensor_info.items()):
     in_print('  outputs[\'%s\'] tensor_info:' % output_key)
-    _print_tensor_info(output_tensor, indent+1)
+    _print_tensor_info(output_tensor, indent + 1)
 
   in_print('Method name is: %s' %
            meta_graph_def.signature_def[signature_def_key].method_name)
 
 
+def _show_defined_functions(saved_model_dir, indent=0):
+  """Prints the function definition of SavedModel2.0 located at saved_model_dir
+
+     Args:
+       saved_model_dir: Directory containing the SavedModel to inspect.
+       indent: How far (in increments of 2 spaces) to indent each line of output.
+  """
+  if context.executing_eagerly():
+    # Disable eager execution to prevent loading of checkpoints
+    ops_lib.disable_eager_execution()
+  trackable_object = load.load(saved_model_dir)
+  indent_str = '  ' * indent
+
+  def in_print(s):
+    print(indent_str + s)
+
+  print('Defined Functions:')
+  functions = save._AugmentedGraphView(
+      trackable_object).list_functions(trackable_object)
+  for name, function in functions.items():
+    in_print('Function Name: \'%s\'' % name)
+    for index, concrete_functions in enumerate(
+            function._list_all_concrete_functions_for_serialization(), 1):
+      args, kwargs = (concrete_functions.structured_input_signature)
```

Comment:
are the brackets needed?

Reply:
https://github.com/tensorflow/tensorflow/blob/59ee7f9138482d85cd93c004aca961bea35820c7/tensorflow/python/saved_model/load.py#L155-L156
I first saw it here. and didn't dare to use something else. the main reason is, sometimes it gives this: ((args), {kwargs}))
and sometimes (args). So I thought may be that's the reason the original author have added the brackets.

Reply:
It should always be (args, kwargs), otherwise it's a bug.

But even if you'd get args, not clear to me, how wrapping into brackets will help.